framework/helpers/docker.py

```diff
@@ -0,0 +1,308 @@
+from datetime import datetime
+from datetime import timedelta
+from math import ceil
+from pathlib import Path
+from pathlib import PosixPath
+from queue import Queue
+from threading import Thread
+from typing import List
+
+from absl import logging
+from docker import DockerClient
+import docker.errors as docker_errors
+import docker.types as docker_types
+import grpc
+from mako.template import Template
+
+from protos.grpc.testing import messages_pb2
+from protos.grpc.testing import test_pb2_grpc
+from protos.grpc.testing.xdsconfig import control_pb2
+from protos.grpc.testing.xdsconfig import service_pb2_grpc
+
+
+class Bootstrap:
+    def __init__(self, base: str, ports: List[int], host_name: str):
```

Comment:
nit: we can use `list[]` typing notation now; no extra imports needed
```suggestion
    def __init__(self, base: str, ports: list[int], host_name: str):
```